### DISCUSSION TITLE
GitHub Pages resets custom domain for deployment every time I push to gh-pages

### DISCUSSION THREAD

**Discussion by kirillbobyrev** [community/community#22366] [Other Feature Feedback, Questions, & Ideas]
<p>I have a fairly simple static website pipeline with Hugo. The <code>main</code> branch has the code, on each push I run the following GitHub Workflow to generate the website and push it to <code>gh-pages</code> branch. Unfortunately, every time a push happens, the “Custom domain” in “Settings” → “Pages” gets reset to an empty one and my website is deployed to <code>username.github.io/project-name</code>, so I have to get back to this setting <em>every time</em> I push some changes. This is extremely inconvenient, why exactly is this happening? Can I stop it from happening in the future?</p>
<p>Here’s what my Workflow looks like.</p>
<pre><code class="lang-yaml">name: Build and deploy
on:
  push:
    branches:
      - main
jobs:
  build:
    name: Build &amp; deploy to GitHub Pages
    runs-on: ubuntu-20.04
    steps:

    - name: Check out code
      uses: actions/checkout@v2
      with:
        submodules: recursive
        fetch-depth: 0

    - name: Setup Hugo
      uses: peaceiris/actions-hugo@v2
      with:
        hugo-version: 'latest'

    - name: Build website
      run: hugo --minify

    - name: Deploy
      uses: peaceiris/actions-gh-pages@v3
      with:
        github_token: ${{ secrets.GITHUB_TOKEN }}
</code></pre>

**Comment by kirillbobyrev** [comment, marked as answer]
<p>Nevermind, I didn’t realize publishing to a custom domain through the settings is just commiting CNAME to gh-pages branch and since I didn’t have static/CNAME in my main branch for Hugo generator, this would be lost after each deployment to gh-pages.</p>

**Reply by dellagustin** [reply]
I had the exact same issue, but with MkDocs, anyway, this helped solving it, thanks for sharing.
Here is my (now fixed) issue: https://github.com/Ecomarx/ecomarx.github.io/issues/5

**Comment by siazon** [comment]
Please add the CNAME file to the public folder. The contents of the CNAME file are your domain name.
